make-hash.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+var intSize = 4
+var zeroBuffer = new Buffer(intSize)
+zeroBuffer.fill(0)
+
+var charSize = 8
+var hashSize = 16
+
+function toArray (buf) {
+  if ((buf.length % intSize) !== 0) {
+    var len = buf.length + (intSize - (buf.length % intSize))
+    buf = Buffer.concat([buf, zeroBuffer], len)
+  }
+
+  var arr = new Array(buf.length >>> 2)
```

Comment:
Bit operations are faster and automatically round into integers

On Tue, Oct 13, 2015, 6:48 PM Daniel Cousens <notifications@github.com>
wrote:

> In make-hash.js
> <https://github.com/crypto-browserify/createHash/pull/9#discussion_r41936608>
> :
>
> > @@ -0,0 +1,30 @@
> > +'use strict'
> > +var intSize = 4
> > +var zeroBuffer = new Buffer(intSize)
> > +zeroBuffer.fill(0)
> > +
> > +var charSize = 8
> > +var hashSize = 16
> > +
> > +function toArray (buf) {
> > +  if ((buf.length % intSize) !== 0) {
> > +    var len = buf.length + (intSize - (buf.length % intSize))
> > +    buf = Buffer.concat([buf, zeroBuffer], len)
> > +  }
> > +
> > +  var arr = new Array(buf.length >>> 2)
>
> buf.length / 4
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crypto-browserify/createHash/pull/9/files#r41936608>.
>